src/services/projects.js

```diff
@@ -64,10 +86,42 @@ angular.module('openshiftCommonServices')
                               });
                     });
           },
+
+          // List the projects the user has access to. This method returns
+          // cached data if the projects had previously been fetched to avoid
+          // requesting them again and again, which is a problem for admins who
+          // might have hundreds or more.
+          list: function(forceRefresh) {
+            if (cachedProjectData && !forceRefresh) {
+              Logger.debug("ProjectsService: returning cached project data");
+              return $q.when(cachedProjectData);
+            }
+
+            Logger.debug("ProjectsService: listing projects, force refresh", forceRefresh);
+            return DataService.list('projects', {}).then(function(projectData) {
+              cachedProjectData = projectData;
+              return projectData;
```

Comment:
OK, so pages that expect to have the latest data would need to use the force flag then. Is there a plan to revisit the pages that are using this function to determine if the projects data could have changed and been the reason for the navigation? For instance, I add to a project and that takes me to another page. That page would expect a list call to return the latest data but it may not if no watcher was set and the force flag was not set.